sdk/src/azure/core/az_json_token.c

```diff
@@ -325,6 +325,52 @@ AZ_NODISCARD static az_result _az_json_token_get_string_helper(
   return AZ_OK;
 }
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ */
+AZ_NODISCARD az_result az_json_string_unescape(
+    az_span json_string,
+    char* destination,
+    int32_t destination_max_size,
+    int32_t* out_string_length)
+{
+  _az_PRECONDITION_VALID_SPAN(json_string, 1, false);
```

Comment:
This is a good precondition. Even though an empty JSON string token is valid (`""`), the caller can check it up front and not have the need to unescape anything in that case.

Reply:
same for a 1 char string , to unescape you need at least 2 chars